tests/preview.rs

```diff
@@ -1,77 +1,191 @@
 #[macro_use]
 extern crate lazy_static;
 
-pub mod utils;
-
-use assert_cmd::prelude::*;
+pub mod fixture;
 
 use std::env;
-use std::fs::File;
-use std::io::Write;
 use std::process::Command;
-use std::str;
 use std::sync::Mutex;
 
+use assert_cmd::prelude::*;
+use fixture::Fixture;
+
 lazy_static! {
     static ref BUILD_LOCK: Mutex<u8> = Mutex::new(0);
 }
 
-macro_rules! settings {
-    ( $f:expr, $x:expr ) => {
-        let file_path = utils::fixture_path($f).join("wrangler.toml");
-        let mut file = File::create(file_path).unwrap();
-        let content = format!(
-            r#"
-            name = "test"
-            zone_id = ""
-            account_id = ""
-            workers_dev = true
-            {}
-        "#,
-            $x
-        );
-        file.write_all(content.as_bytes()).unwrap();
-    };
-}
-
 #[test]
 fn it_can_preview_js_project() {
-    let fixture = "simple_js";
-    utils::create_temporary_copy(fixture);
-    settings! {fixture, r#"
+    let fixture = Fixture::new("simple_js");
+    fixture.create_file(
+        "index.js",
+        r#"
+        addEventListener('fetch', event => {
+            event.respondWith(handleRequest(event.request))
+        })
+        
+        /**
+        * Fetch and log a request
+        * @param {Request} request
+        */
+        async function handleRequest(request) {
+            return new Response('Hello worker!', { status: 200 })
+        }
+    "#,
+    );
+    fixture.create_default_package_json();
+    fixture.create_wrangler_toml(
+        r#"
         type = "javascript"
-    "#};
+    "#,
+    );
     preview(fixture);
-    utils::cleanup(fixture);
 }
 
 #[test]
 fn it_can_preview_webpack_project() {
-    let fixture = "webpack_simple_js";
-    utils::create_temporary_copy(fixture);
-    settings! {fixture, r#"
+    let fixture = Fixture::new("webpack_simple_js");
+    fixture.scaffold_webpack();
+    fixture.create_wrangler_toml(
+        r#"
         type = "webpack"
-    "#};
+    "#,
+    );
```

Comment:
do we also want to call "cleanup" at the end of every one of these tests? I feel like that would be advisable.

Reply:
he added it to the `preview` helper function: https://github.com/cloudflare/wrangler/pull/854/files#diff-eb986e3abcfb440d6233373e1d67f78dR190

Reply:
gotcha! I would prefer for cleanup to explicitly happen as its own call instead of being put in preview(), given that preview() should not have cleanup as a side effect. OR change preview() to preview_and_cleanup().

Reply:
fwiw, as part of my test refactor work i will likely update fixtures to conditionally clean up based on whether tests pass, per the test fixture library used by wasm_pack.

Reply:
Doesn't make sense for me to call cleanup separately I think - `preview` will both create the files and clean them up. We don't actually create the files until we enter the `preview` function even though we call `fixture.createFile` that's just a string in memory